### PR TITLE
Add taup-wavefront tutorial

### PR DESCRIPTION
**Pull request by seisman**
Address #1.

### CODE REVIEW

File: source/taup/taup_wavefront.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+.. image:: taup_wavefront.jpg
+   :width: 500 px
+   :align: center
+
+进阶用法
+--------
+
+``taup wavefront`` 提供了 ``--rays`` 和 ``--timestep`` 选项进一步控制生成的波前面时
+所使用的射线数目（即每个波前面时用多少个点约束）和时间步长（即间隔多长时间产生一个
+波前面）。
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:66]:
```suggestion
``taup wavefront`` 提供的 ``--rays`` 选项可以进一步控制生成波前面时所使用的射线数目
（即每个波前面用多少个点约束），而 ``--timestep`` 选项可以时间步长
（即间隔多长时间产生一个波前面）。
```

File: source/taup/taup_wavefront.rst [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+下面的示例设置了每个波前面用20条射线计算，且每25秒计算一个波前面::
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:68]:
```suggestion
以下示例中，每个波前面用 20 条射线计算，且每 25 秒计算一个波前面::
```

File: source/taup/taup_wavefront.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+- 震中距（度）
+- 深度（km）
+- 走时（秒）
+- 射线参数（秒/度）
+
+可以直接使用文件的前两列进行绘图。
+
+加上 ``-gmt`` 选项，则会生成一个可用于绘制波前面传播的 GMT 脚本 :file:`taup_wavefront.gmt`::
+
+    $ taup wavefront -ph P -deg 60 -h 300 -gmt
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:49]:
```suggestion
加上 ``--gmt`` 选项，则会生成一个可用于绘制波前面传播的 GMT 脚本 :file:`taup_wavefront.gmt`::

    $ taup wavefront -ph P -deg 60 -h 300 --gmt
```

File: source/taup/taup_wavefront.rst [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+.. image:: taup_wavefront_ray20-time25.jpg
+   :width: 500 px
+   :align: center
+
+波场动画
+--------
+
+有了每个时刻的波场信息，就可以制作波场动画了。
+
+``taup wavefront`` 提供了 ``--timefiles`` 选项，可以将每个时刻的波前面信息输出到
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:84]:
```suggestion
``taup wavefront`` 提供的 ``--timefiles`` 选项可以将每个时刻的波前面信息输出到
```

File: source/taup/taup_wavefront.rst [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+不同的文件中。例如::
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:85]:
```suggestion
不同的文件中。

以下命令会生成几十个文件，每个文件包含一个波前面信息::
```

File: source/taup/taup_wavefront.rst [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    $ taup wavefront -ph P -deg 60 -h 300 --timestep 10 --timefiles
+
+以上命令会生成几十个文件，每个文件包含一个波前面信息。例如 :file:`taup_wavefront_025.gmt`
+包含了第 25 秒时的波前面信息。这些文件即可用于制图并进一步生成动画。
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:90]:
```suggestion
例如，\ :file:`taup_wavefront_025.gmt` 包含了第 25 秒时的波前面信息。
这些文件即可用于制图并进一步生成动画。
```

File: source/taup/taup_wavefront.rst [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+将 ``--timefiles`` 选项与 ``-gmt`` 选项连用，则会对每个波前面生成一个单独的 GMT 绘图脚本。
+
+    $ taup wavefront -ph P -deg 60 -h 300 --timestep 10 --timefiles --gmt
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:94]:
```suggestion
将 ``--timefiles`` 选项与 ``--gmt`` 选项连用，则会对每个波前面生成一个单独的 GMT 绘图脚本::

    $ taup wavefront -ph P -deg 60 -h 300 --timestep 10 --timefiles --gmt
```

File: source/taup/taup_wavefront.rst [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+以上命令生成了几十个 GMT 脚本。可以使用如下命令批量执行所有 GMT 脚本并生成多个 PDF 文件::
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:96]:
```suggestion
以上命令会生成几十个 GMT 脚本。使用如下命令可以批量执行所有 GMT 脚本，生成多个 PDF 文件::
```

File: source/taup/taup_wavefront.rst [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+
+    $ for i in $(ls taup_wavefront_*.gmt); do bash $i; done
+
+使用 `GraphicsMagick 的 gm 命令 <http://www.graphicsmagick.org/convert.html>`_
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:100]: Could we use two links? one for GraphicsMagick, the other is for `gm`.

File: source/taup/taup_wavefront.rst [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+可以将多个 PDF 文件转换为 GIF 动画::
+
+    $ gm convert -delay 10 -density 150x150 taup_wavefront_*.pdf taup_wavefront.gif
+
+其中 ``-delay 10`` 表示每两张图的间隔是 10/100 秒，\ ``-density 150x150`` 设置动画的水平和
+垂直方向的像素点数。
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:106]:
```suggestion
其中，\ ``-delay 10`` 表示每两张图的间隔是 10/100 秒，\ ``-density 150x150`` 设置动画的水平和
垂直方向的像素点数。
```

File: source/taup/taup_wavefront.rst (hunk 9 above)
Comment (seisman) [inline source/taup/taup_wavefront.rst:100]:
```suggestion
使用 `GraphicsMagick <http://www.graphicsmagick.org/>`_ 的 
`gm convert <http://www.graphicsmagick.org/convert.html>`_ 命令
```

File: source/taup/taup_wavefront.rst [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+--------
+
+使用 ``taup wavefront --help`` 可查询该命令的完整用法和选项。其选项与
+:doc:`taup_time` 几乎完全相同，故不再重复介绍。此处仅介绍常用用法。
+
+计算震源深度为 300 km，震中距 60° 处，P 震相的波前面::
+
+    $ taup wavefront -ph P -deg 60 -h 300
+
+该命令会生成文件 :file:`taup_wavefront.gmt`\ ，其内容如下::
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:22]:
```suggestion
该命令会生成文件 :file:`taup_wavefront.gmt`\ ，其包含了不同时刻（t=100, 200, 300, ...）
P 波的波前面信息，内容如下::
```

File: source/taup/taup_wavefront.rst [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+    ...(省略若干行)
+    > P at 200.0 seconds
+    15.69    4244.7   200.00  254.331
+    15.93    4267.0   200.00  260.031
+    16.10    4283.9   200.00  264.250
+    16.23    4296.4   200.00  267.326
+    ...(省略若干行)
+
+该文件中包含了不同时刻（t=100, 200, 300, ...）P 波的波前面信息。输出包含四列，
+分别是：
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:38]:
```suggestion
输出包含四列，分别是：震中距（度）、深度（km）、走时（秒）、射线参数（秒/度）。可以直接使用文件的前两列进行绘图。
```

File: source/taup/taup_wavefront.rst [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+
+- 震中距（度）
+- 深度（km）
+- 走时（秒）
+- 射线参数（秒/度）
+
+可以直接使用文件的前两列进行绘图。
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:45]:
```suggestion
```

File: source/taup/taup_wavefront.rst (hunk 1 above)
Comment (core-man) [reply source/taup/taup_wavefront.rst:66]: I am still not clear about `--rays` option.

File: source/taup/taup_wavefront.rst [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+
+使用 ``taup wavefront --help`` 可查询该命令的完整用法和选项。其选项与
+:doc:`taup_time` 几乎完全相同，故不再重复介绍。此处仅介绍常用用法。
+
+计算震源深度为 300 km，震中距 60° 处，P 震相的波前面::
+
+    $ taup wavefront -ph P -deg 60 -h 300
+
+该命令会生成文件 :file:`taup_wavefront.gmt`\ ，其包含了不同时刻（t=100, 200, 300, ...）
+P 波的波前面信息，内容如下::
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:23]:
```suggestion
的 P 波的波前面信息，内容如下::
```

File: source/taup/taup_wavefront.rst [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+    5.55    5167.7   100.00  267.326
+    ...(省略若干行)
+    > P at 200.0 seconds
+    15.69    4244.7   200.00  254.331
+    15.93    4267.0   200.00  260.031
+    16.10    4283.9   200.00  264.250
+    16.23    4296.4   200.00  267.326
+    ...(省略若干行)
+
+输出包含四列，分别是：震中距（度）、深度（km）、走时（秒）、射线参数（秒/度）。可以直接使用文件的前两列进行绘图。
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:38]:
```suggestion
该文件包含四列，分别是：震中距（度）、深度（km）、走时（秒）、射线参数（秒/度）。可以直接使用文件的前两列进行绘图。
```

File: source/taup/taup_wavefront.rst (hunk 1 above)
Comment (seisman) [reply source/taup/taup_wavefront.rst]: Let's say you want to manually calculate the wavefront for t=100 s. How to do that? You can set up 10 rays with different parameters, calculate the ray path of each ray and determine the traveltimes at each position along the ray path. With 10 rays, you can get 10 positions for t=100 s, and the 10 positions can be used to draw the wavefront.

File: source/taup/taup_wavefront.rst [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+
+加上 ``--gmt`` 选项，则会生成一个可用于绘制波前面传播的 GMT 脚本 :file:`taup_wavefront.gmt`::
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:40]:
```suggestion
加上 ``--gmt`` 选项，则会生成一个绘制波前面的 GMT 脚本 :file:`taup_wavefront.gmt`::
```

File: source/taup/taup_wavefront.rst [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+
+.. image:: taup_wavefront.jpg
+   :width: 500 px
+   :align: center
+
+进阶用法
+--------
+
+``taup wavefront`` 提供的 ``--rays`` 选项可以进一步控制生成波前面时所使用的射线数目
+（即每个波前面用多少个点约束），而 ``--timestep`` 选项可以时间步长
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:58]:
```suggestion
（即每个波前面用多少个点约束），而 ``--timestep`` 选项可以控制波前面的时间步长
```

File: source/taup/taup_wavefront.rst [hunk 18]
```diff
@@ -0,0 +1,108 @@
...
+   :width: 500 px
+   :align: center
+
+波场动画
+--------
+
+有了每个时刻的波场信息，就可以制作波场动画了。
+
+``taup wavefront`` 提供的 ``--timefiles`` 选项可以将每个时刻的波前面信息输出到
+不同的文件中。
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:78]:
```suggestion
``taup wavefront`` 提供的 ``--timefiles`` 选项可以将每个时刻的波前面信息输出到
不同的文件中。有了每个时刻的波场信息，就可以制作波场动画了。
```

File: source/taup/taup_wavefront.rst [hunk 19]
```diff
@@ -0,0 +1,108 @@
...
+
+以下命令会生成几十个文件，每个文件包含一个波前面信息::
+
+    $ taup wavefront -ph P -deg 60 -h 300 --timestep 10 --timefiles
+
+例如，\ :file:`taup_wavefront_025.gmt` 包含了第 25 秒时的波前面信息。
+这些文件即可用于制图并进一步生成动画。
```
Comment (core-man) [inline source/taup/taup_wavefront.rst:85]:
```suggestion
这些文件可用于制图，并进一步生成动画。
```

File: source/taup/taup_wavefront.rst (hunk 1 above)
Comment (core-man) [reply source/taup/taup_wavefront.rst]: Clear.

File: source/taup/taup_wavefront.rst [hunk 20]
```diff
@@ -0,0 +1,106 @@
...
+
+使用 ``taup wavefront --help`` 可查询该命令的完整用法和选项。其选项与
+:doc:`taup_time` 几乎完全相同，故不再重复介绍。此处仅介绍常用用法。
+
+计算震源深度为 300 km，震中距 60° 处，P 震相的波前面::
+
+    $ taup wavefront -ph P -deg 60 -h 300
+
+该命令会生成文件 :file:`taup_wavefront.gmt`\ ，其包含了不同时刻（t=100, 200, 300, ...）
+P 波的波前面信息，内容如下::
```
Comment (seisman) [inline source/taup/taup_wavefront.rst:23]:
```suggestion
该命令会生成文件 :file:`taup_wavefront.gmt`\ ，其包含了不同时刻（t=100, 200, 300, ...）的
P 波的波前面信息，内容如下::
```

**Review by core-man — APPROVED** [review]

**Comment by core-man** [comment]
Why do we remove Maggie from https://seismo-learn.org/links/tools/?

**Pull request merged**